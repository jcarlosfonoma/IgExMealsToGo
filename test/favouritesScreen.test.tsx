import React from "react"
import { create } from "react-test-renderer"
import { FavouritesScreen } from "../app/features/settings/screens/favourites.screen"
import { FavouritesContextProvider } from "../app/services/favourites/favourites.context"
import { AuthenticationContextProvider } from "../app/services/authentication/authentication.context"
import { ThemeProvider } from "styled-components/native"
import { theme } from "../app/infrastructure/theme"

const navigation = {
  navigate: jest.fn(),
}

const tree = create(
  <ThemeProvider theme={theme}>
    <AuthenticationContextProvider>
      <FavouritesContextProvider>
        <FavouritesScreen navigation={navigation} />
      </FavouritesContextProvider>
    </AuthenticationContextProvider>
  </ThemeProvider>,
)

jest.mock("@expo/vector-icons", () => ({
  Ionicons: "",
}))

jest.mock("firebase", () => {
  const firebaseObject = {
    onAuthStateChanged: () => {
      return null
    },
  }

  return {
    auth: () => {
      return { ...firebaseObject }
    },
  }
})

describe("Favourites screen tests", () => {
  test("should match snapshot", () => {
    expect(tree).toMatchSnapshot()
  })
})
