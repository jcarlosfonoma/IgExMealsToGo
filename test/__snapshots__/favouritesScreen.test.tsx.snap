// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites screen tests should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    center={true}
    style={
      Array [
        Object {
          "color": "#262626",
          "flexWrap": "wrap",
          "fontFamily": "Oswald_400Regular",
          "fontSize": 16,
          "fontWeight": "400",
          "marginBottom": 0,
          "marginTop": 0,
        },
      ]
    }
    variant="body"
  >
    No favourites yet
  </Text>
</RCTSafeAreaView>
`;
